tests/backup/backup_helper.go

```diff
@@ -1426,3 +1435,200 @@ func DeleteBackupAndWait(backupName string, ctx context.Context) error {
 	_, err := task.DoRetryWithTimeout(backupDeletionSuccessCheck, backupDeleteTimeout, backupDeleteRetryTime)
 	return err
 }
+
+// GetPxBackupVersion returns the version of Px Backup like 2.4.0-e85b680
+func GetPxBackupVersion() (string, error) {
+	ctx, err := backup.GetAdminCtxFromSecret()
+	if err != nil {
+		return "", err
+	}
+	versionResponse, err := Inst().Backup.GetPxBackupVersion(ctx, &api.VersionGetRequest{})
+	if err != nil {
+		return "", err
+	}
+	version := versionResponse.GetVersion()
+	return fmt.Sprintf("%s.%s.%s-%s", version.GetMajor(), version.GetMinor(), version.GetPatch(), version.GetGitCommit()), nil
+}
+
+// GetPxBackupVersionSemVer returns the version of Px Backup in semver format like 2.4.0
+func GetPxBackupVersionSemVer() (string, error) {
+	ctx, err := backup.GetAdminCtxFromSecret()
+	if err != nil {
+		return "", err
+	}
+	versionResponse, err := Inst().Backup.GetPxBackupVersion(ctx, &api.VersionGetRequest{})
+	if err != nil {
+		return "", err
+	}
+	version := versionResponse.GetVersion()
+	return fmt.Sprintf("%s.%s.%s", version.GetMajor(), version.GetMinor(), version.GetPatch()), nil
+}
+
+// GetPxBackupBuildDate returns the Px Backup build date
+func GetPxBackupBuildDate() (string, error) {
+	ctx, err := backup.GetAdminCtxFromSecret()
+	if err != nil {
+		return "", err
+	}
+	versionResponse, err := Inst().Backup.GetPxBackupVersion(ctx, &api.VersionGetRequest{})
+	if err != nil {
+		return "", err
+	}
+	version := versionResponse.GetVersion()
+	return version.GetBuildDate(), nil
+}
```

Comment:
Wouldn't it be cleaner if move getting version into it's own function:
```go
func GetPxBackupVersion() (*api.Version......, error) {
	ctx, err := backup.GetAdminCtxFromSecret()
	if err != nil {
		return "", err
	}
	versionResponse, err := Inst().Backup.GetPxBackupVersion(ctx, &api.VersionGetRequest{})
	if err != nil {
		return "", err
	}
	version := versionResponse.GetVersion()
	return version
}
```
and then have two functions:
```go
func GetPxBackupVersionString() (string, error) {
	version, err := GetPxBackupVersion()
	if err != nil {
		return "", err
	}
	return fmt.Sprintf("%s.%s.%s-%s", version.GetMajor(), version.GetMinor(), version.GetPatch(), version.GetGitCommit()), nil
}
```
and
```go
func GetPxBackupVersionSemVer() (string, error) {
	version, err := GetPxBackupVersion()
	if err != nil {
		return "", err
	}
	return fmt.Sprintf("%s.%s.%s", version.GetMajor(), version.GetMinor(), version.GetPatch()), nil
}
```
and maybe not have `GetPxBackupBuildDate` as it's simply gonna be doing `version.GetBuildDate()` for anyone using it.

Reply:
Fixed based on the first part of your comment but for the second part

`version.GetBuildDate()` is not that intuitive when compared to having a function with name `GetPxBackupBuildDate`
Which is why I thought of going this way.